jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -358,22 +358,27 @@ private void buildAndCacheApplicationLayers() {
   private void buildImage() {
     ProgressEventDispatcher.Factory childProgressDispatcherFactory =
         Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
-
-    results.builtImage =
+    results.builtImages =
         executorService.submit(
-            () ->
-                new BuildImageStep(
-                        buildContext,
-                        childProgressDispatcherFactory,
-                        results.baseImagesAndRegistryClient.get().images.get(0),
-                        realizeFutures(
-                            Verify.verifyNotNull(
-                                results
-                                    .baseImagesAndLayers
-                                    .get()
-                                    .get(results.baseImagesAndRegistryClient.get().images.get(0)))),
-                        realizeFutures(Verify.verifyNotNull(results.applicationLayers)))
-                    .call());
+            () -> {
+              List<Future<Image>> builtImages = new ArrayList<>();
+              for (Image image : results.baseImagesAndLayers.get().keySet()) {
+                Future<Image> builtImage =
+                    executorService.submit(
+                        () ->
+                            new BuildImageStep(
+                                    buildContext,
+                                    childProgressDispatcherFactory,
+                                    image,
+                                    realizeFutures(
+                                        Verify.verifyNotNull(
+                                            results.baseImagesAndLayers.get().get(image))),
+                                    realizeFutures(Verify.verifyNotNull(results.applicationLayers)))
+                                .call());
+                builtImages.add(builtImage);
+              }
```

Comment:
You can use `.entrySet()` to iterator over a Map. An entry is a (key, value) pair, so no need to call `Map.get(key)` to get the value.

```suggestion
              for (Map.Entry<Image, List<Future<PreparedLayer>>> entry :
                  results.baseImagesAndLayers.get().entrySet()) {
                Future<Image> builtImage =
                    executorService.submit(
                        () ->
                            new BuildImageStep(
                                    buildContext,
                                    childProgressDispatcherFactory,
                                    entry.getKey(), // base image
                                    realizeFutures(entry.getValue()), // base image layers
                                    realizeFutures(Verify.verifyNotNull(results.applicationLayers)))
                                .call());
                builtImages.add(builtImage);
              }
```